File: src/test/java/Example1.java
import java.io.Serializable;
import java.util.function.Consumer;

public class Example1 {

  public static Consumer<Object> exampleLambda() {
    return (x) -> { };
  }

  public interface SerializableConsumer extends Serializable {
    void consume();
  }

  public static SerializableConsumer exampleLambda2() {
    return () -> { };
  }
}

File: src/test/java/example1.kt
import org.junit.Test

class Example1Test {

  @Test
  fun testExampleLambda1() {
    val example1 = Example1.exampleLambda()
    printTypeInfo(example1)
  }

  @Test
  fun testExampleLambda2() {
    val example2 = Example1.exampleLambda2()
    printTypeInfo(example2)
  }

}
